### PR TITLE
Auth

### LINKED ISSUE #27: Firebase Authentication

### PR DESCRIPTION
**Pull request by Rofy87**
1- Added missing pacakage react-redux and store data of user in the redux store
2- login with facebook , google and normal signup 
3- logout

signup

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ function App() {
       <Route path="/" component={NavBar} />
       <Route exact path="/" component={Homepage} />
       <Route path="/resources" component={ResourcesPage} />
-      <Route exact path="/about" component={AboutUs} />
```
Comment (halitbatur) [inline src/App.js:17]: why are these removed

File: src/components/Login/Login.js [hunk 2]
```diff
@@ -1,13 +1,53 @@
...
+  signInWithEmailAndPassword,
+} from "../../helpers/auth";
 
 function Login({ show, hideFn }) {
+  const [errorMsg, setErrorMsg] = useState(false);
+  const dispatch = useDispatch();
+
+  const signWithSocial = async (social) => {
+    let fnCall = null;
+    if (social === "google") fnCall = signInWithGoogle;
```
Comment (halitbatur) [inline src/components/Login/Login.js]: can we have curly brackets instead of no brackets

File: src/components/Login/Login.js [hunk 3]
```diff
@@ -1,13 +1,53 @@
...
+    else if (social === "facebook") fnCall = signInWithFacebook;
+    if (fnCall) {
+      const result = await fnCall();
+      if (result.error) {
+        setErrorMsg("Something went wrong");
+      } else {
+        setErrorMsg(false);
+        dispatch({ type: SIGN_IN, payload: result });
+        hideFn(false);
```
Comment (halitbatur) [inline src/components/Login/Login.js]: what is hideFn here can we have a more relevant name to it ?

File: src/components/Login/Login.js [hunk 4]
```diff
@@ -1,13 +1,53 @@
...
+      }
+    }
+  };
+
+  const signIn = async (data, setSubmitting) => {
+    const result = await signInWithEmailAndPassword(data);
+    if (result.error) {
+      if (result.error.customError) setErrorMsg(result.error.customError);
```
Comment (halitbatur) [inline src/components/Login/Login.js]: no one line if statements please use curly brackets

File: src/components/SignUpModal/SignUpModal.js [hunk 5]
```diff
@@ -1,43 +1,97 @@
...
+  registerWithEmailAndPassword,
+} from "../../helpers/auth";
 
 function SignUpModal({ show, hideFn }) {
+  const [errorMsg, setErrorMsg] = useState(false);
+  const dispatch = useDispatch();
+
+  const signWithSocial = async (social) => {
+    let fnCall = null;
+    if (social === "google") fnCall = signInWithGoogle;
```
Comment (halitbatur) [inline src/components/SignUpModal/SignUpModal.js]: same comments here with the curly brackets

**Review by halitbatur — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 1 above)
Comment (Rofy87) [reply src/App.js:17]: it was not removed it was modified to eliminate the "exact" attribute

File: src/components/Login/Login.js (hunk 3 above)
Comment (Rofy87) [reply src/components/Login/Login.js]: Done

File: src/components/Login/Login.js (hunk 2 above)
Comment (Rofy87) [reply src/components/Login/Login.js]: Done

File: src/App.js (hunk 1 above)
Comment (Rofy87) [reply src/App.js:17]: Done

File: src/components/Login/Login.js (hunk 4 above)
Comment (Rofy87) [reply src/components/Login/Login.js]: Done

File: src/components/SignUpModal/SignUpModal.js (hunk 5 above)
Comment (Rofy87) [reply src/components/SignUpModal/SignUpModal.js]: Done

Comment (Rofy87) [reply src/components/SignUpModal/SignUpModal.js]: I also added the onAuthStateChanged for managing the refresh bug

**Review by halitbatur — APPROVED** [review]

File: src/components/Login/Login.js [hunk 6]
```diff
@@ -1,25 +1,80 @@
...
 import * as Yup from "yup";
+import { useDispatch } from "react-redux";
+import { SIGN_IN } from "../../store/actions/actionConst";
+import {
+  signInWithGoogle,
+  signInWithFacebook,
+  signInWithEmailAndPassword,
+} from "../../helpers/auth";
+
+function Login({ show, showModal }) {
```
Comment (halitbatur) [inline src/components/Login/Login.js]: show is still not a good variable name, for booleans please use the keyword is for example isModalShown implies that the value if either true or false

File: src/components/Login/Login.js [hunk 7]
```diff
@@ -1,25 +1,80 @@
...
+  const [errorMsg, setErrorMsg] = useState(false);
+  const dispatch = useDispatch();
+
+  const signWithSocial = async (social) => {
+    let fnCall = null;
```
Comment (halitbatur) [inline src/components/Login/Login.js]: this is kinda useless since the signWithSocial will always have either a google or facebook

File: src/components/Login/Login.js [hunk 8]
```diff
@@ -1,25 +1,80 @@
...
+    if (social === "google") {
+      fnCall = signInWithGoogle;
+    } else if (social === "facebook") {
+      fnCall = signInWithFacebook;
+    }
+    if (fnCall) {
```
Comment (halitbatur) [inline src/components/Login/Login.js]: looking are previous comment that means that we dont need this if condidtion

**Review by halitbatur — CHANGES_REQUESTED** [review]

File: src/components/Login/Login.js (hunk 8 above)
Comment (Rofy87) [reply src/components/Login/Login.js]: Done

File: src/components/Login/Login.js (hunk 7 above)
Comment (Rofy87) [reply src/components/Login/Login.js]: Done

File: src/components/Login/Login.js (hunk 6 above)
Comment (Rofy87) [reply src/components/Login/Login.js]: Done

**Review by ayse8888 — APPROVED** [review]
looks good over all to me. great work Rahaf, thank you. Halit's requested changes have been fixed as well.

**Review by mustafaguldag — APPROVED** [review]
It looks solid and no unnesesary spaces. Thanks Rahaf

File: src/helpers/auth.js [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+      const profile = db
+        .collection("profile")
+        .where("uid", "==", user.uid)
+        .get();
+      return profile.docs;
+    } else {
+      return false;
+    }
+  });
+  // const user = await auth.onAuthStateChanged();
```
Comment (halitbatur) [inline src/helpers/auth.js]: no commented code please

**Review by halitbatur — APPROVED** [review]
just remove commented code and it can be merged

File: src/helpers/auth.js (hunk 9 above)
Comment (Rofy87) [reply src/helpers/auth.js]: Done
